aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -17,22 +17,31 @@ namespace c10 {
  */
 class OpKernel final {
 public:
-  explicit constexpr OpKernel(KernelFunction* kernel): kernel_(kernel) {}
+  explicit OpKernel(KernelFunction* kernel, KernelStateCreatorFunction* state_creator)
+  : kernel_(kernel), state_creator_(state_creator) {}
 
   OpKernel(OpKernel&&) = default;
   OpKernel& operator=(OpKernel&&) = default;
   OpKernel(const OpKernel&) = delete;
   OpKernel& operator=(const OpKernel&) = delete;
 
-  IValue call(ArrayRef<IValue> args) const {
-    return (*kernel_)(args);
+  /**
+   * Call the operator kernel with the given arguments.
+   */
+  IValue call(ArrayRef<IValue> args) {
+    if (state_.get() == nullptr) {
+      state_ = (*state_creator_)();
```

Comment:
Don't you have to null pointer check here for `state_creator_`?

Reply:
Added an assertion. `state_creator_` is never null, even if the kernel has no state.